stdlib/jms/src/main/ballerina/jms/queue_receiver.bal

```diff
@@ -119,19 +117,17 @@ public type QueueReceiverActions object {
 
 function QueueReceiverActions.receiveFrom(Destination destination, int timeoutInMilliSeconds = 0) returns (Message|
         error)? {
-    match (self.queueReceiver) {
-        QueueReceiver queueReceiver => {
-            match (queueReceiver.config.session) {
-                Session s => {
-                    validateQueue(destination);
-                    queueReceiver.createQueueReceiver(s, queueReceiver.config.messageSelector, destination = destination
-                    );
-                }
-                () => {}
-            }
-        }
-        () => { log:printInfo("Message receiver is not properly initialised for queue " +
-                destination.destinationName); }
+    var queueReceiver = self.queueReceiver;
+    if (queueReceiver is QueueReceiver) {
+          var session = queueReceiver.config.session;
+          if (session is Session) {
+              validateQueue(destination);
+              queueReceiver.createQueueReceiver(session, queueReceiver.config.messageSelector, destination = destination);
+          } else {
+
+          }
+    } else if (queueReceiver is ()) {
```

Comment:
Shall we have only the else clause instead of having type-test expression for `()`? Because within that else-if clause, particular variable is not used. So we can reduce the implicit casting cost there.
```ballerina
else {
   log:printInfo("Message receiver is not properly initialised for queue " + destination.destinationName);
}
```
Apply the same to other similar occurrences where type tested variable is not used.

Reply:
Resolved